### PR TITLE
🐛Syncer: Remaining fixes on downstream namespace cleaning

### PR DESCRIPTION
**Pull request by davidfestal**
## Summary

This Remaining fixes on downstream namespace cleaning in the Syncer

## Related issues

No dedicated issue.
This fixes issues found in the code when working on issue https://github.com/kcp-dev/kcp/issues/2358

### CODE REVIEW

File: pkg/syncer/spec/spec_controller.go [hunk 1]
```diff
@@ -168,10 +169,14 @@ func NewSpecSyncer(syncerLogger logr.Logger, syncTargetWorkspace logicalcluster.
...
 						// Use namespace lister
 						nsObj, err := namespaceLister.Get(namespace)
-						if err != nil {
+						if err != nil && !errors.IsNotFound(err) {
 							utilruntime.HandleError(err)
 							return
 						}
+						if errors.IsNotFound(err) {
+							return
+						}
```
Comment (ncdc) [inline pkg/syncer/spec/spec_controller.go:178]:
```suggestion
						if errors.IsNotFound(err) {
							return
						}
						if err != nil {
							utilruntime.HandleError(err)
							return
						}
```

Would this be cleaner?

Comment (ncdc) [reply pkg/syncer/spec/spec_controller.go:178]: Or
```go
if err != nil {
  if !errors.IsNotFound(err) {
    utilruntime.HandleError(err)
  }
  return
}
```

Comment (davidfestal) [reply pkg/syncer/spec/spec_controller.go:178]: Sure.

**Comment by davidfestal** [comment]
@ncdc The `e2e-tests-ppc64le` failure doesn't seem related to the code, but to infrastructure problems (afaict)

**Comment by ncdc** [comment]
@davidfestal that's not a required test, so it won't block merging.

@jmprusi can you please review this?

**Comment by davidfestal** [comment]
@jmprusi is on vacation this week.

**Pull request merged**